sdk/instrumentation/net/http/handler.go

```diff
@@ -87,16 +88,25 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 		defer r.Body.Close()
 
+		isMultipartFormDataBody := HasMultiPartFormDataContentTypeHeader(headerMapAccessor{r.Header})
+
 		// Only records the body if it is not empty and the content type
 		// header is not streamable
 		if len(body) > 0 {
-			setTruncatedBodyAttribute("request", body, int(h.dataCaptureConfig.BodyMaxSizeBytes.Value), span)
+			setTruncatedBodyAttribute("request", body, int(h.dataCaptureConfig.BodyMaxSizeBytes.Value), span,
+				isMultipartFormDataBody)
 		}
 
 		processingBody := body
 		if int(h.dataCaptureConfig.BodyMaxProcessingSizeBytes.Value) < len(body) {
 			processingBody = body[:h.dataCaptureConfig.BodyMaxProcessingSizeBytes.Value]
 		}
+		// if body is multipart/form-data, base64 encode it before passing it on to the filter
+		if isMultipartFormDataBody {
```

Comment:
@tyroneyu does nginx also have processing size?

Reply:
Yes, and the usage is the same as here.  It's the maximum size of the original body that we will take to process.  (not the base64 encoded string).